.github/scripts/need-respin.sh

```diff
@@ -0,0 +1,22 @@
+#! /bin/bash
+set -euxo pipefail
+
+image=$1
+#  Get Latest Keycloak release tag
+tag=$(curl -sL http://github.com/keycloak/keycloak/releases/latest -H "Accept: application/json" | jq -r '.tag_name')
+
+docker pull ${image}:${tag}
+
+base_image_url=$(docker inspect ${image}:${tag} | jq -r '.[0].Config.Labels.url')
+
+base_image_name=$(docker inspect ${image}:${tag} | jq -r '.[0].Config.Labels.url' | sed 's/.*#\///' | sed 's/\/images.*//')
+
+docker pull ${base_image_name}:latest
+
+latest_base_image_url=$(docker inspect ${base_image_name}:latest | jq -r '.[0].Config.Labels.url')
```

Comment:
It is possible to access the manifests without downloading the images. It's a little tricky, as each repo seems to implement the dockerhttp API slightly differently (particularly around the access tokens), but it could save a lot of time.

Quay needs an anonymous token with the right service and scope parameters:

```bash
host=quay.io
repository=keycloak/keycloak
tag=18.0.0
token="$(curl -sSfL 'https://$host/v2/auth?service=$host&scope=repository:$repository:pull' | jq -r '.token')"
curl -sSfL -H "Authorization: Bearer $token" "https://$host/v2/$repository/manifests/$tag" | jq -r '.history[0].v1Compatibility' | jq -r '.config.Labels.url'
```

Which returns `https://access.redhat.com/containers/#/registry.access.redhat.com/ubi8-minimal/images/8.6-751`

Red Hat's repo provides tokenless anonymous access:
```
```bash
host=registry.access.redhat.com
repository=ubi8-minimal
tag=latest
curl -sSfL "https://$host/v2/$repository/manifests/$tag" | jq -r '.history[0].v1Compatibility' | jq -r '.config.Labels.url'
```

Which returns `https://access.redhat.com/containers/#/registry.access.redhat.com/ubi8-minimal/images/8.6-751`

So you could make your approach to the comparison faster based of off this method. It won't be as generic an approach as you're trying to write. However, the assumption around the url label isn't generic either, it's something that happens to work for UBI

That got me thinking, this check could also be done by comparing the layer hashes. The code for that is simpler, compared to parsing the label, but does require you to hardcode the parent URL:

```bash
curl -sSfL -H 'Authorization: Bearer $token' https://quay.io/v2/keycloak/keycloak/manifests/18.0.0 | jq -r '.fsLayers | .[] | .blobSum' > /tmp/layers-kc
curl -sSfL https://registry.access.redhat.com/v2/ubi8-minimal/manifests/latest | jq -r '.fsLayers | .[] | .blobSum' > /tmp/layers-ubi
grep -qxF -f /tmp/layers-kc /tmp/layers-ubi
```

Which would return exit code 0 if the latest ubi layers were present in the keycloak layers (no respin required), or exit code 1 if they were not (respin required)

Reply:
Right, it's a tradeoff :)

I think for a once-per-day check, the performance hit of downloading several hundred megabytes of data shouldn't be too bad